stripecardscan/src/main/java/com/stripe/android/stripecardscan/cardscan/CardScanActivity.kt

```diff
@@ -0,0 +1,683 @@
+        when (newState) {
+            is ScanState.NotFound -> {
+                viewBinding.viewFinderBackground
+                    .setBackgroundColor(getColorByRes(R.color.stripeNotFoundBackground))
+                viewBinding.viewFinderWindow
+                    .setBackgroundResource(R.drawable.stripe_card_background_not_found)
+                viewBinding.viewFinderBorder.startAnimation(R.drawable.stripe_card_border_not_found)
+                viewBinding.instructions.setText(R.string.stripe_card_scan_instructions)
+            }
+            is ScanState.FoundShort -> {
+                viewBinding.viewFinderBackground
+                    .setBackgroundColor(getColorByRes(R.color.stripeFoundBackground))
+                viewBinding.viewFinderWindow
+                    .setBackgroundResource(R.drawable.stripe_card_background_found)
+                viewBinding.viewFinderBorder.startAnimation(R.drawable.stripe_card_border_found)
+                viewBinding.instructions.setText(R.string.stripe_card_scan_instructions)
+                viewBinding.instructions.show()
+            }
+            is ScanState.FoundLong -> {
+                viewBinding.viewFinderBackground
+                    .setBackgroundColor(getColorByRes(R.color.stripeFoundBackground))
+                viewBinding.viewFinderWindow
+                    .setBackgroundResource(R.drawable.stripe_card_background_found)
+                viewBinding.viewFinderBorder
+                    .startAnimation(R.drawable.stripe_card_border_found_long)
+                viewBinding.instructions.setText(R.string.stripe_card_scan_instructions)
+                viewBinding.instructions.show()
+            }
+            is ScanState.Correct -> {
+                viewBinding.viewFinderBackground
+                    .setBackgroundColor(getColorByRes(R.color.stripeCorrectBackground))
+                viewBinding.viewFinderWindow
+                    .setBackgroundResource(R.drawable.stripe_card_background_correct)
+                viewBinding.viewFinderBorder.startAnimation(R.drawable.stripe_card_border_correct)
+                viewBinding.instructions.hide()
+            }
+        }
+    }
+
+    private fun changeScanState(newState: ScanState): Boolean {
+        if (newState == scanStatePrevious || scanStatePrevious?.isFinal == true) {
+            return false
+        }
+
+        scanState = newState
+        displayState(newState, scanStatePrevious)
+        scanStatePrevious = newState
+        return true
+    }
+
+    private val scanErrorListener = object : AnalyzerLoopErrorListener {
+        override fun onAnalyzerFailure(t: Throwable): Boolean {
+            Log.e(Config.logTag, "Error executing analyzer", t)
+            return false
+        }
+
+        override fun onResultFailure(t: Throwable): Boolean {
+            Log.e(Config.logTag, "Error executing result", t)
+            return true
+        }
+    }
```

Comment:
Can we move this into a separate shared class between the CardScanActivity and CardImageVerificationActivity?